crates/stackable-versioned/tests/basic.rs

```diff
@@ -5,14 +5,16 @@ use stackable_versioned::Versioned;
 #[versioned(
     version(name = "v1alpha1"),
     version(name = "v1beta1"),
-    version(name = "v1")
+    version(name = "v1"),
+    version(name = "v2"),
+    version(name = "v3")
 )]
 struct Foo {
     /// My docs
     #[versioned(
         added(since = "v1alpha1"),
         renamed(since = "v1beta1", from = "jjj"),
-        deprecated(since = "v1", _note = "")
+        deprecated(since = "v2", note = "not empty")
```

Comment:
As mentioned in Slack my biggest concern is that we are missing `#[deprecated = "not empty"]` on the top level `Foo` (version v3) struct. I think I would prefer to generate a `v3::Foo` struct and not have any special case handling for the latest version.

But you are much more knowledgeable on this, so I would leave it to you

![image](https://github.com/stackabletech/operator-rs/assets/29303194/670483c3-2d82-401a-b656-f8623e7d9d32)

Reply:
This is indeed a very tricky situation. Some of the reasons why this is so tricky are:

- We want the "base" struct (where all the version and action definition takes place) to always represent the latest version.
- If we generate a `v3::Foo`, the base struct will never be used - leading to dead code. There might be a solution similar to our CRD Spec generation. But that needs to be explored.

I will explore this in more detail in the next few days with a bunch of experiments on "real" CRDs. For now, I would leave it as is and would propose that I will pick this up as my next PR.